rocketpy/motors/Tank.py

```diff
@@ -0,0 +1,626 @@
+        """
+        centerOfMass = (
+            self.liquidCenterOfMass * self.liquidMass
+            + self.gasCenterOfMass * self.gasMass
+        ) / (self.mass)
+
+        # Check for zero mass
+        bound_mass = self.mass < 0.001 * self.geometry.total_volume * self.gas.density
+        if bound_mass.any():
+            # TODO: pending Function setter impl.
+            centerOfMass.yArray[bound_mass] = self.geometry.bottom
+            centerOfMass.setInterpolation()
+            centerOfMass.setExtrapolation()
+
+        return centerOfMass
+
+    @funcify_method("Time (s)", "Inertia tensor of liquid (kg*m²)")
+    def liquidInertiaTensor(self):
+        """
+        Returns the inertia tensor of the liquid portion of the tank
+        as a function of time. The reference point is the center of
+        mass of the tank.
+
+        Returns
+        -------
+        rocketpy.Function
+            Inertia tensor of the liquid portion of the tank as a
+            function of time.
+        """
+        Ix_volume = self.geometry.Ix_volume(self.geometry.bottom, self.liquidHeight.max)
+        Ix_volume = Ix_volume @ self.liquidHeight
+
+        # Steiner theorem to account for center of mass
+        Ix_volume -= self.liquidVolume * self.liquidCenterOfMass**2
+        Ix_volume += (
+            self.liquidVolume * (self.liquidCenterOfMass - self.centerOfMass) ** 2
+        )
+
+        return self.liquid.density * Ix_volume
+
+    @funcify_method("Time (s)", "inertia tensor of gas (kg*m^2)")
+    def gasInertiaTensor(self):
+        """
+        Returns the inertia tensor of the gas portion of the tank
+        as a function of time. The reference point is the center of
+        mass of the tank.
+
+        Returns
+        -------
+        rocketpy.Function
+            Inertia tensor of the gas portion of the tank as a
+            function of time.
+        """
+        Ix_volume = self.geometry.Ix_volume(self.geometry.bottom, self.gasHeight.max)
+        lower_inertia_volume = Ix_volume @ self.liquidHeight
+        upper_inertia_volume = Ix_volume @ self.gasHeight
+        inertia_volume = upper_inertia_volume - lower_inertia_volume
+
+        # Steiner theorem to account for center of mass
+        inertia_volume -= self.gasVolume * self.gasCenterOfMass**2
+        inertia_volume += (
+            self.gasVolume * (self.gasCenterOfMass - self.centerOfMass) ** 2
+        )
+
+        return self.gas.density * inertia_volume
+
+    @funcify_method("Time (s)", "inertia tensor (kg*m^2)")
+    def inertiaTensor(self):
+        """
+        Returns the inertia tensor of the tank's fluids as a function of
+        time. The reference point is the center of mass of the tank.
+
+        Returns
+        -------
+        Function
+            Inertia tensor of the tank's fluids as a function of time.
+        """
+        return self.liquidInertiaTensor + self.gasInertiaTensor
+
+
+class MassFlowRateBasedTank(Tank):
+    def __init__(
```

Comment:
Docstring please 🥲

Reply:
At a first look, I wanted to understand what is the actual difference of using one tank model or another. Should they always give me the same results?

Reply:
Docstrings added on 80daa46310121108ea6557f44e5c4663c2892ef3. The difference and assumptions behind each Tank were succinctly described in the added docstrings. 

To sum up, the UllageBasedTank and LevelBasedTank are less accurate than the others, since they assume uniform gas distribution filling all the portion that is not occupied by liquid. Therefore, these tanks can only model the tank state until the liquid runs out.